### PR TITLE
feature/refactor_TCE-771_TCE-735_TCE-779

### PR DESCRIPTION
**Pull request by codacy-vrhpires**
- added plugin react-refresh
- security and security-node had discrepancies in categorization and severity
- fixed bug related to how a config file (in the repo) was detected and used
- reverted _.eslintrc_ config file support deprecation
- styling patterns were [deprecated](https://github.com/eslint/eslint/issues/17522) in @typecript-eslint [package](https://www.npmjs.com/package/@stylistic/eslint-plugin) and moved to a separate package
    - a new parsing method for those rules was implemented because it didn’t follow current pattern _/docs/rules/*.md_
- [node-fetch](https://www.npmjs.com/package/node-fetch) was [converted to ESM](https://github.com/node-fetch/node-fetch/blob/main/docs/v3-UPGRADE-GUIDE.md#converted-to-es-module) so its usage was replaced by [axios](https://www.npmjs.com/package/axios) and the package removed
- docs/description folder is now emptied every time documentation is generated to purge deprecated patterns description files
- updated scripts section in _package.json_ to follow current styling
    - added _eslint_ script (and styling rules) to fix the tool code styling automatically
    - added _upgrade_ script to upgrade and install all packages and then generate docs in a single command
- improved _dockerfile_ efficiency
- added test to assert _description.json_ and _patterns.json_ are created
- refactored _docGenerator_ class to streamline its usage in _docGeneratorMain_
- removed need to parse _package.json_ to retrieve Eslint version in _toolmetada_

### CODE REVIEW

**Review by stefanvacareanu7 — DISMISSED** [review]
I see a lot of removed patterns. Was this intended? Reading the description, not all of those are included.

Other than this, LGTM!

File: Dockerfile [hunk 1]
```diff
@@ -1,26 +1,25 @@
 FROM node:lts-alpine3.19 as builder
 
-COPY . ./
+COPY src ./src
```
Comment (mrfyda) [inline Dockerfile]: to benefit from the docker build cache you probably want to move the src after installing the dependencies:

```
COPY package.json .
COPY package-lock.json .
COPY tsconfig.json .

RUN npm install --legacy-peer-deps --omit=dev

COPY src ./src
COPY docs ./docs

RUN npm cache clean --force &&\
    npm run build:docs &&\
    npm run test
    ```

File: Dockerfile [hunk 2]
```diff
@@ -1,26 +1,25 @@
...
+COPY package.json .
+COPY package-lock.json .
+COPY tsconfig.json .
 
 RUN npm install --legacy-peer-deps --omit=dev &&\
     npm cache clean --force &&\
-    npm run compile &&\
-    npm test &&\
-    npm run build:docs
+    npm run build:docs &&\
```
Comment (mrfyda) [inline Dockerfile]: does this also build the eslint app? (that we then copy from /dist)

Comment (codacy-vrhpires) [reply Dockerfile]: yes. To make sure the latest changes are in place before docs generation it always builds first.

> "build": "tsc --showConfig && tsc",

> "build:docs": "npm run build && node dist/src/docGeneratorMain.js",

File: src/test/docGenerator.spec.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import * as fs from "node:fs"
+
+import {equal} from "assert"
+
+describe("DocGenerator", () => {
```
Comment (mrfyda) [inline src/test/docGenerator.spec.ts:5]: now that the files exist in the repo, if the DocGenerator fails, this test will continue to pass

**Review by stefanvacareanu7 — DISMISSED** [review]

File: src/test/docGenerator.spec.ts (hunk 3 above)
Comment (codacy-vrhpires) [reply src/test/docGenerator.spec.ts:5]: the aim of the test is to make sure that after emptying the docs/description folder it also writes the new file:
```typescript
    await this.emptyDocsDescriptionFolder()
    await this.writeFileInJson(
      this.docsDescriptionDirectory + "description.json",
      descriptions
    )
```
if DocGenerator fails the file will most likely still exist and it's fine that it passes.

**Review by mrfyda — APPROVED** [review]

**Pull request merged**